bios/main.c

```diff
@@ -168,38 +136,50 @@ static uint32_t copy_dtb(uint32_t dst, uint32_t src)
 static void copy_ns_images(void)
 {
 	uint32_t dst;
+	long r;
 
 	/* 32MiB above beginning of RAM */
 	kernel_entry = DRAM_START + 32 * 1024 * 1024;
 
 	/* Copy non-secure image in place */
-	dst = copy_bios_image("kernel", kernel_entry, &__linker_nsec_blob_start,
-			&__linker_nsec_blob_end);
+	r = semihosting_download_file("zImage", 64 * 1024 * 1024, kernel_entry);
+	CHECK(r < 0);
+	dst = kernel_entry + r;
 
 	dtb_addr = ROUNDUP(dst, PAGE_SIZE) + 96 * 1024 * 1024; /* safe spot */
 	dst = copy_dtb(dtb_addr, DTB_START);
 
 	rootfs_start = ROUNDUP(dst + DTB_MAX_SIZE, PAGE_SIZE);
```

Comment:
minor: (might be out-of-scope of this p-r ?) `dst` is already increased by `DTB_MAX_SIZE`.